ui/webui/test/README.rst

```diff
@@ -1,17 +1,46 @@
-===================================
+Anaconda Web UI tests
+=====================
 
-This directory contains automated integration tests for Anaconda WebUI, and the support files for them.
+The Web UI tests are based on cockpit's testing framework which in turn is based
+on python's unittest library. There are two kinds of Web UI tests, "unit" that
+tests only one specific feature at a time and "integration" that goes through the
+entire installation process including reboot to the installed system. Both types
+use the same library of helper functions, to avoid duplicating code and to make
+the test development easier, but they are executed in different ways because of
+the higher requirements of the integration tests.
 
+Test development
+----------------
+
+Before test case for testing a new user interface can be written any new elements
+in the UI that are going to be interacted with need to be covered by helper
+functions. These helper functions are located in ``./ui/webui/test/helpers`` and
+are organized by screens or installations steps. These helper functions are then
+used in both unit and integration tests.
+
+For interaction with elements on page use class `Browser <https://github.com/cockpit-project/cockpit/blob/a19bb79512b4816d65dee4923fe00ef4b0217cdd/test/common/testlib.py#L182>`_.
+For running commands in the installation environment use class `Machine <https://github.com/cockpit-project/bots/blob/1df595efa53fbf02731108d7a3657642d5b92c9e/machine/machine_core/machine.py#L55>`_ / `SSHConnection <https://github.com/cockpit-project/bots/blob/1df595efa53fbf02731108d7a3657642d5b92c9e/machine/machine_core/ssh_connection.py#L45>`_.
+
+All helper functions should be decorated with `log_step <https://github.com/rhinstaller/anaconda/blob/9baed94ac63fb2da73231ccc27ed88ac4360858b/ui/webui/test/helpers/step_logger.py#L11>`_.
+That makes sure the call and parameters are logged in the test output. The decorator
+also has options to create browser snapshots (screenshot and html) and those should
+be used whenever the helper function is interacting with the UI.
+
+Web UI unit tests
+=================
+
+Unit tests are stored in directory `./ui/webui/test`, in files named `check-{something}`.
+
+For information about the @nondestructive decorator and some best practices read `Cockpit's test documentation <https://github.com/cockpit-project/cockpit/tree/main/test/#guidelines-for-writing-tests>`_.
 Before running the tests refer to the ``CONTRIBUTING`` guide in the root of the repository for installation of all the necessary build and test dependencies.
 
 Preparation and general invocation
 ----------------------------------
 
 *Warning*: Never run the build, test, or any other command here as root!
 
-To run the WebUI integration tests run the following from the root of the anaconda repo.
-(do NOT run the integration tests as root).
+To run the WebUI tests run the following from the root of the anaconda repo.
+(do NOT run the tests as root).
```

Comment:
I would remove the warning line, and let natural selection decide whose laptop lives. 

(joking, but I mean it, just not this PR as it did not introduce it)